pkgs/stdenv/generic/setup.sh

```diff
@@ -802,14 +802,17 @@ dumpVars() {
 # Utility function: echo the base name of the given path, with the
 # prefix `HASH-' removed, if present.
 stripHash() {
-    local strippedName
+    local strippedName casematchOpt=0
```

Comment:
I think you can use ```oldOpts=$(shopt -p nocasematch)``` to match how runHook works above.

Reply:
Then ```eval $oldOpts``` at the end

Reply:
I can do that, but this way is faster. I realize that the performance difference hardly matters here, but it just bugged me to fork the shell unnecessarily.

If I'm going to accept the cost of a fork, then it would actually be simpler overall to skip this and do

```bash
if ( shopt -u nocasematch; [[ "$strippedName" =~ ^[a-z0-9]{32}- ]] ); then
```

which just unsets it in the subshell.

Reply:
FWIW, I measured it and `shopt -s nocasematch; shopt -u nocasematch` is somewhere around 80x faster (on my machine) than `( shopt -s nocasematch )`. More generally (again on my machine), `shopt -s nocasematch; shopt -u nocasematch` takes <10µs whereas spawning a subshell takes at least 650µs.

This seems like a tiny difference, but imagine if I'm running `stripHash` on 1000 files. That 650µs turns into 650ms. And we're already spawning a separate process for `basename` as it is.

For that matter, I'm tempted to replace the call to `basename` with more string manipulation. It seems relatively straightforward except for one thing, which is that the behavior of `basename //` is documented as being platform-specific, depending on whether the platform in question treats `//` as semantically distinct from `/`. I don't know what platforms do this (does Cygwin do this?) but the inability to replicate this edge case makes me leery about replacing the call.